commit 5352d29817bce351042d9a99061f4ccd96f441cc
Author: Tomas Pizarro <tompizmor@users.noreply.github.com>
Date:   Wed Apr 4 14:08:08 2018 +0200

    [stable/redmine] Refactor image property in values.yaml (#4657)
    
    * [stable/redmine] Refactor image properties in values.yaml
    
    * Use rolling tag and refactor pullSecrets
    
    * Fix lint issue
    
    * Update Chart.yaml

diff --git a/stable/redmine/Chart.yaml b/stable/redmine/Chart.yaml
index a159c7d03..77d63030e 100644
--- a/stable/redmine/Chart.yaml
+++ b/stable/redmine/Chart.yaml
@@ -1,5 +1,5 @@
 name: redmine
-version: 2.0.6
+version: 3.0.0
 appVersion: 3.4.4
 description: A flexible project management web application.
 keywords:
diff --git a/stable/redmine/README.md b/stable/redmine/README.md
index abd124b09..8b0b67429 100644
--- a/stable/redmine/README.md
+++ b/stable/redmine/README.md
@@ -53,37 +53,40 @@ helm install --name my-release stable/redmine --set databaseType.mariadb=false,d
 
 The following table lists the configurable parameters of the Redmine chart and their default values.
 
-| Parameter | Description | Default |
-| --------- | ----------- | ------- |
-| `image` | Redmine image | `bitnami/redmine:{VERSION}` |
-| `imagePullPolicy` | Image pull policy | `IfNotPresent` |
-| `redmineUsername` | User of the application | `user` |
-| `redminePassword` | Application password | _random 10 character long alphanumeric string_ |
-| `redmineEmail` | Admin email | `user@example.com` |
-| `redmineLanguage` | Redmine default data language | `en` |
-| `extraVars` | Environment variables, passed to redmine | `nil` |
-| `smtpHost` | SMTP host | `nil` |
-| `smtpPort` | SMTP port | `nil` |
-| `smtpUser` | SMTP user | `nil` |
-| `smtpPassword` | SMTP password | `nil` |
-| `smtpTls` | Use TLS encryption with SMTP | `nil` |
-| `databaseType.postgresql` | Select postgresql database | `false` |
-| `databaseType.mariadb` | Select mariadb database | `true` |
-| `mariadb.mariadbRootPassword` | MariaDB admin password | `nil` |
-| `postgresql.postgresqlPassword` | PostgreSQL admin password | `nil` |
-| `serviceType` | Kubernetes Service type | `LoadBalancer` |
-| `serviceLoadBalancerSourceRanges` | An array of load balancer sources | `0.0.0.0/0` |
-| `ingress.enabled` | Enable or disable the ingress | `false` |
-| `ingress.hostname` | The virtual host name | `redmine.cluster.local` |
-| `ingress.annotations` | An array of service annotations | `nil` |
-| `ingress.tls[i].secretName | The secret kubernetes.io/tls | `nil` |
-| `ingress.tls[i].hosts[j] | The virtual host name | `nil` |
-| `networkPolicyApiVersion` | The kubernetes network API version | `extensions/v1beta1` |
-| `persistence.enabled` | Enable persistence using PVC | `true` |
-| `persistence.existingClaim` | The name of an existing PVC | `nil` |
-| `persistence.storageClass` | PVC Storage Class | `nil` (uses alpha storage class annotation) |
-| `persistence.accessMode` | PVC Access Mode | `ReadWriteOnce` |
-| `persistence.size` | PVC Storage Request | `8Gi` |
+|            Parameter              |              Description                 |                          Default                        | 
+| --------------------------------- | ---------------------------------------- | ------------------------------------------------------- |
+| `image.registry`                  | Redmine image registry                   | `docker.io`                                             |
+| `image.repository`                | Redmine image name                       | `bitnami/redmine`                                       |
+| `image.tag`                       | Redmine image tag                        | `{VERSION}`                                             |
+| `image.pullPolicy`                | Image pull policy                        | `Always` if `imageTag` is `latest`, else `IfNotPresent` |
+| `image.pullSecrets`               | Specify image pull secrets               | `nil`                                                   |
+| `redmineUsername`                 | User of the application                  | `user`                                                  |
+| `redminePassword`                 | Application password                     | _random 10 character long alphanumeric string_          |
+| `redmineEmail`                    | Admin email                              | `user@example.com`                                      |
+| `redmineLanguage`                 | Redmine default data language            | `en`                                                    |
+| `extraVars`                       | Environment variables, passed to redmine | `nil`                                                   |
+| `smtpHost`                        | SMTP host                                | `nil`                                                   |
+| `smtpPort`                        | SMTP port                                | `nil`                                                   |
+| `smtpUser`                        | SMTP user                                | `nil`                                                   |
+| `smtpPassword`                    | SMTP password                            | `nil`                                                   |
+| `smtpTls`                         | Use TLS encryption with SMTP             | `nil`                                                   |
+| `databaseType.postgresql`         | Select postgresql database               | `false`                                                 |
+| `databaseType.mariadb`            | Select mariadb database                  | `true`                                                  |
+| `mariadb.mariadbRootPassword`     | MariaDB admin password                   | `nil`                                                   |
+| `postgresql.postgresqlPassword`   | PostgreSQL admin password                | `nil`                                                   |
+| `serviceType`                     | Kubernetes Service type                  | `LoadBalancer`                                          |
+| `serviceLoadBalancerSourceRanges` | An array of load balancer sources        | `0.0.0.0/0`                                             |
+| `ingress.enabled`                 | Enable or disable the ingress            | `false`                                                 |
+| `ingress.hostname`                | The virtual host name                    | `redmine.cluster.local`                                 |
+| `ingress.annotations`             | An array of service annotations          | `nil`                                                   |
+| `ingress.tls[i].secretName`       | The secret kubernetes.io/tls             | `nil`                                                   |
+| `ingress.tls[i].hosts[j]`         | The virtual host name                    | `nil`                                                   |
+| `networkPolicyApiVersion`         | The kubernetes network API version       | `extensions/v1beta1`                                    |
+| `persistence.enabled`             | Enable persistence using PVC             | `true`                                                  |
+| `persistence.existingClaim`       | The name of an existing PVC              | `nil`                                                   |
+| `persistence.storageClass`        | PVC Storage Class                        | `nil` (uses alpha storage class annotation)             |
+| `persistence.accessMode`          | PVC Access Mode                          | `ReadWriteOnce`                                         |
+| `persistence.size`                | PVC Storage Request                      | `8Gi`                                                   |
 
 The above parameters map to the env variables defined in [bitnami/redmine](http://github.com/bitnami/bitnami-docker-redmine). For more information please refer to the [bitnami/redmine](http://github.com/bitnami/bitnami-docker-redmine) image documentation.
 
diff --git a/stable/redmine/templates/deployment.yaml b/stable/redmine/templates/deployment.yaml
index 739beecda..51c2a65da 100644
--- a/stable/redmine/templates/deployment.yaml
+++ b/stable/redmine/templates/deployment.yaml
@@ -14,10 +14,16 @@ spec:
       labels:
         app: {{ template "redmine.fullname" . }}
     spec:
+      {{- if .Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range .Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end}}
+      {{- end }}
       containers:
       - name: {{ template "redmine.fullname" . }}
-        image: "{{ .Values.image }}"
-        imagePullPolicy: {{ default "" .Values.imagePullPolicy | quote }}
+        image: "{{ .Values.image.registry }}/{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy | quote }}
         env:
         {{- if .Values.extraVars }}
 {{ toYaml .Values.extraVars | indent 8 }}
diff --git a/stable/redmine/values.yaml b/stable/redmine/values.yaml
index 329541552..4718d8789 100644
--- a/stable/redmine/values.yaml
+++ b/stable/redmine/values.yaml
@@ -1,12 +1,21 @@
 ## Bitnami Redmine image version
 ## ref: https://hub.docker.com/r/bitnami/redmine/tags/
 ##
-image: bitnami/redmine:3.4.4-r2
-
-## Specify a imagePullPolicy
-## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
-##
-imagePullPolicy: IfNotPresent
+image:
+  registry: docker.io
+  repository: bitnami/redmine
+  tag: 3.4.4
+  ## Specify a imagePullPolicy
+  ## Defaults to 'Always' if image tag is 'latest', else set to 'IfNotPresent'
+  ## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
+  ##
+  pullPolicy: IfNotPresent
+  ## Optionally specify an array of imagePullSecrets.
+  ## Secrets must be manually created in the namespace.
+  ## ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
+  ##
+  # pullSecrets:
+  #   - myRegistrKeySecretName
 
 ## User of the application
 ## ref: https://github.com/bitnami/bitnami-docker-redmine/#environment-variables
